### ISSUE TITLE
[RFC] AST structure of TSImportType to be more align with rest of AST

### ISSUE THREAD

**Issue by armano2** [typescript-eslint/typescript-eslint#2998]
**THIS IS BREAKING CHANGE**

### What is this change about?

`TSImportType` has boolean property `isTypeOf` that is inconsistent with other usages of `typeof` within AST structure.

`isTypeOf` property is unique (not used anywhere else in code)

```ts
type x = typeof import('x')
```
```
TSImportType:
  - isTypeOf: boolean
```

```ts
type x = typeof window
```
```
TSTypeQuery:
  - exprName: Identifier
```

### Why we should change it?

- Improve consistency of ast structure.
- Align generated structure with babel representation
- Allow consumers (eg. prettier) to correctly place comment - [link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEMCeAHOACAHtgXm3SwgDNsB6AKm0gFt6EZtrLsBLejCAJxgAUAclxCAlCAA0ICBhgdoAZ2SgAhr14QA7gAV1CZSlUAbLarTLpAI16qwAazgwAyhjscoAc2QxeAVzhpOHorOAATMPCAGVUvP1VPOAAxPnpVGHkvZBBVPxgIKRAACxh6YwB1Io54RTcwOGcDao4AN2q0bLBFSxAPRTh+HVtPNOQyE37pACtFXAAhWwcnZ1UmKI84MYnAkBncZw9PYzgART8IeC3jSZA3Xn7ebJI4RTBeDjlCjHfYco4wmBFZAADgADNJvhB+uVbBhst8XgMWptpABHc7wIayQw5RQAWigcHC4UKvDg6I4ZKGCVGSHG1x2-XoHB8-kZh2OZwumzp22kMFUVj+AKBSAATPzbBxjIcAMIQRiqbIvACshT8-QAKoLDPSbi0AgBJKCRWDON4fGAAQRNznQxyu-QAvk6gA)
  ```ts
  typeof /* comment */ import('x')
  ```

**Comment by JoshuaKGoldberg** [comment]
#3076 is merged into #5886; closing this issue now accordingly. It'll be released as a part of the v6 major release.
